runtime/compiler/net/ClientStream.cpp

```diff
@@ -0,0 +1,278 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "ClientStream.hpp"
+#include "control/Options.hpp"
+#include "env/VerboseLog.hpp"
+#include "net/SSLProtobufStream.hpp"
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <netinet/tcp.h>	/* for tcp_nodelay option */
+#include <fcntl.h>
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h> /// gethostname, read, write
+#if defined(JITSERVER_ENABLE_SSL)
+#include <openssl/err.h>
+#include "control/CompilationRuntime.hpp"
+#endif
+
+namespace JITServer
+{
+int ClientStream::_numConnectionsOpened = 0;
+int ClientStream::_numConnectionsClosed = 0;
+
+// used for checking server compatibility
+int ClientStream::_incompatibilityCount = 0;
+uint64_t ClientStream::_incompatibleStartTime = 0;
+const uint64_t ClientStream::RETRY_COMPATIBILITY_INTERVAL = 10000; //ms
```

Comment:
Rather than the 'ms' comment here, renaming the field to `RETRY_COMPATIBILITY_INTERVAL_MS` would make it clear everywhere.